Sources/Sentry/SentryProfiler.mm

```diff
@@ -109,9 +108,8 @@
         const auto v2LifecycleIsTrace = profileIsContinuousV2 && v2LifecycleValue != nil
             && v2Lifecycle == SentryProfileLifecycleTrace;
         const auto profileIsCorrelatedToTrace = !profileIsContinuousV2 || v2LifecycleIsTrace;
-        SentryUIViewControllerPerformanceTracker *performanceTracker =
-            [SentryDependencyContainer.sharedInstance uiViewControllerPerformanceTracker];
-        if (profileIsCorrelatedToTrace && performanceTracker.alwaysWaitForFullDisplay) {
+
+        if (profileIsCorrelatedToTrace && options.enableTimeToFullDisplayTracing) {
```

Comment:
I think we'll also need to check a few other options, like `options.enableUIViewControllerTracing`, since you could theoretically misconfigure the SDK like:
```
options.enableUIViewControllerTracing = false
options.enableTimeToFullDisplayTracing = true
```

Checking `SentryUIViewControllerPerformanceTracker.alwaysWaitForFullDisplay` would have implicitly included all the option combination validation. I don't know what other options would be applicable, maybe `enableAutoPerformanceTracing` and `tracesSampleRate`/`tracesSampler`?

Reply:
This is actually why the UI tests are failing now. Since we automatically configure `enableUIViewControllerTracing` to `true` unless the launch flag `"--disable-time-to-full-display-tracing"` is set in SentrySDKWrapper/SentrySDKOverrides, changing this now caused the conditional branch to be taken here where it was not before.